src/worker.py

```
@@ -1416,3 +1443,125 @@ async def on_fetch(request, env):
     except Exception as e:
         await capture_exception(e, request, env, "on_fetch_unhandled")
         return err("Internal server error", 500)
+
+
+# ---------------------------------------------------------------------------
+# Notifications API
+# ---------------------------------------------------------------------------
+
+async def _create_notification(env, user_id: str, type_: str, title: str,
+                                message: str, related_id: str = None) -> None:
```

Ruff (0.15.10)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)